syncany-lib/src/main/java/org/syncany/operations/up/UpOperation.java

```diff
@@ -106,145 +119,301 @@ public UpOperation(Config config, UpOperationOptions options) {
 		this.options = options;
 		this.result = new UpOperationResult();
 		this.localDatabase = new SqlDatabase(config);
-		this.remoteTransaction = new RemoteTransaction(config, transferManager);
 	}
 
 	@Override
-	public UpOperationResult execute() throws Exception {
+	public UpOperationResult execute() throws OperationException {
 		logger.log(Level.INFO, "");
 		logger.log(Level.INFO, "Running 'Sync up' at client " + config.getMachineName() + " ...");
 		logger.log(Level.INFO, "--------------------------------------------");
 
 		fireStartEvent();
+		try {
```

Comment:
- \n before 'try'
- This function is now becoming rather ugly. All the try/catches don't make it better. I would love to see some refactoring on that (mainly splitting into smaller functions) -- again, not in this pull though